packages/core/src/RenderingEngine/StackViewport.ts

```diff
@@ -1745,113 +1754,103 @@ class StackViewport extends Viewport implements IStackViewport {
+      error,
+      imageIdIndex,
+      imageId,
+    };
 
-      function sendRequest(imageId, imageIdIndex, options) {
-        return loadAndCacheImage(imageId, options).then(
-          (image) => {
-            successCallback.call(this, image, imageIdIndex, imageId);
-          },
-          (error) => {
-            errorCallback.call(this, error, imageIdIndex, imageId);
-          }
-        );
-      }
+    triggerEvent(eventTarget, Events.IMAGE_LOAD_ERROR, eventDetail);
+  }
 
-      /**
-       * If use16bittexture is specified, the CSWIL will automatically choose the
-       * array type when no targetBuffer is provided. When CSWIL is initialized,
-       * the use16bit should match the settings of cornerstone3D (either preferSizeOverAccuracy
-       * or norm16 textures need to be enabled)
-       *
-       * If use16bittexture is not specified, we force the Float32Array for now
-       */
-      const priority = -5;
-      const requestType = RequestType.Interaction;
-      const additionalDetails = { imageId };
-      const options = {
-        targetBuffer: {
-          type: this.useNativeDataType ? undefined : 'Float32Array',
-        },
-        preScale: {
-          enabled: true,
-        },
-        useRGBA: false,
-      };
+  public getTargetOptions(imageId: string) {
+    const { transferSyntaxUID } = metaData.get('transferSyntax', imageId) || {};
 
-      const eventDetail: EventTypes.PreStackNewImageEventDetail = {
-        imageId,
-        imageIdIndex,
-        viewportId: this.id,
-        renderingEngineId: this.renderingEngineId,
-      };
-      triggerEvent(this.element, Events.PRE_STACK_NEW_IMAGE, eventDetail);
+    /**
+     * If use16bittexture is specified, the CSWIL will automatically choose the
+     * array type when no targetBuffer is provided. When CSWIL is initialized,
+     * the use16bit should match the settings of cornerstone3D (either preferSizeOverAccuracy
+     * or norm16 textures need to be enabled)
+     *
+     * If use16bittexture is not specified, we force the Float32Array for now
+     */
+    const options = {
+      targetBuffer: {
+        type: this.useNativeDataType ? undefined : 'Float32Array',
+      },
+      preScale: {
+        enabled: true,
+      },
+      useRGBA: false,
+      transferSyntaxUID,
+    };
+    return options;
+  }
 
-      imageLoadPoolManager.addRequest(
-        sendRequest.bind(this, imageId, imageIdIndex, options),
-        requestType,
-        additionalDetails,
-        priority
-      );
+  private _loadAndDisplayImageGPU(imageId: string, imageIdIndex: number) {
+    const eventDetail: EventTypes.PreStackNewImageEventDetail = {
+      imageId,
+      imageIdIndex,
+      viewportId: this.id,
+      renderingEngineId: this.renderingEngineId,
+    };
+    triggerEvent(this.element, Events.PRE_STACK_NEW_IMAGE, eventDetail);
+
+    return progressiveLoader.loadSingle(imageId, this).then((v) => {
+      return imageId;
     });
```

Comment:
So i just noticed that we are not using the imageLoadPoolManager anymore. is that true? what will happen to our queue?

Reply:
I believe that progressive loading should not be imposed on everyone for every use case. Instead, we should consider it as a configuration option in the cornerstone3D initialization settings. This way, people won't encounter unexpected results with their backends (although there may still be bugs). We should strive to explain the benefits of progressive loading and encourage users to try it. Eventually, we can make it the default setting.